lib/ruby/lib/alces/stack/boot/run.rb

```diff
@@ -88,7 +91,9 @@ def save_template(parameters={})
           add_kickstart(parameters) if @kickstart_template
           combiner = Alces::Stack::Templater::Combiner.new(@json, parameters)
           save = get_save_file(combiner)
-          @to_delete << save
+          if !@permanent_boot_flag
```

Comment:
Generally better to use `unless` rather than `if !`; ref: https://github.com/bbatsov/ruby-style-guide#unless-for-negatives

Also, favour the modifier form of `if`/`unless` for single lines; ref: https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier

i.e.:

```
@to_delete << save unless @permanent_boot_flag
```